### PR TITLE
Pacman Game

### LINKED ISSUE #809: Pacman Game
# Aim

- To create a basic pacman game using python

# Details

- The script will have basic things like any other pacman game, a maze which has some aliens or monsters and obstacles. The objective of pacman is to come out of the maze successfully.

### PR DESCRIPTION
**Pull request by sayantani11**
# Description

This is a simple pacman game, just like original.
There are 4 enemies and one pacman, the task is to get as many coins as possible within 3 chances.

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings

### CODE REVIEW

File: Pacman-Game/Pacman.py [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+        font = pygame.font.SysFont('arial black', 16, bold = True, italic = True)
+        text = font.render(words, False, colour)
+        text_size = text.get_size()
+        if centered:
+            pos[0] = pos[0]-text_size[0]//2
+            pos[1] = pos[1]-text_size[1]//2
+        screen.blit(text, pos)
+        
+    def load(self):
+        self.background = pygame.image.load('maze.png')
```
Comment (kaustubhgupta) [inline Pacman-Game/Pacman.py:284]:
```suggestion
        self.background = pygame.image.load('./Pacman-Game/maze.png')
```

File: Pacman-Game/Pacman.py [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+        self.background = pygame.transform.scale(self.background, (MAZE_WIDTH,MAZE_HEIGHT))
+        # File that contains walls
+        with open("walls.txt",'r') as file:
```
Comment (kaustubhgupta) [inline Pacman-Game/Pacman.py:287]:
```suggestion
        with open("./Pacman-Game/walls.txt",'r') as file:
```

**Review by kaustubhgupta — COMMENTED** [review]
These are your errors returned by bot:

**Comment by sayantani11** [comment]
okay I will check the errors today itself

**Comment by sayantani11** [comment]
@kaustubhgupta I tried fixing the errors what are the errors now.....The problem is I can only run the code in jupyter lab and sometimes it avoids catching the errors

**Comment by kaustubhgupta** [comment]
> @kaustubhgupta I tried fixing the errors what are the errors now.....The problem is I can only run the code in jupyter lab and sometimes it avoids catching the errors

Hmm, you don't have any IDE setup like VSCODE or Pycharm (Python-specific)? If not then you should at least setup vscode as it will be beneficial for your future projects as well!

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
The game is not starting. It terminates as soon as the pygame messge in the terminal comes:
```
pygame 1.9.6
Hello from the pygame community. https://www.pygame.org/contribute.html
```

**Comment by sayantani11** [comment]
> > @kaustubhgupta I tried fixing the errors what are the errors now.....The problem is I can only run the code in jupyter lab and sometimes it avoids catching the errors
> 
> Hmm, you don't have any IDE setup like VSCODE or Pycharm (Python-specific)? If not then you should at least setup vscode as it will be beneficial for your future projects as well!

I have both PyCharm and VScode but the problem is I guess the path where I downloaded a few of the python packages, and the path of VScode or Pycharm isn't same

**Comment by sayantani11** [comment]
I am looking for the issue.... I will try fixing it

**Comment by kaustubhgupta** [comment]
@sayantani11 updates?

**Comment by sayantani11** [comment]
@kaustubhgupta I am working on it ......Give me today's time.... If I am unable then I will inform

**Comment by sayantani11** [comment]
@kaustubhgupta I guess this won't fix

**Comment by kaustubhgupta** [comment]
@sayantani11 oh okay no issue, you can work on other issues now

**Pull request closed without merging**